src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -141,6 +142,21 @@ RebuildQueryStrings(Job *workerJob)
 								? "(null)"
 								: TaskQueryString(task))));
 
+		Datum partitionColumnValue;
+		Oid partitionColumnType = 0;
+		char *partitionColumnString = NULL;
+		if (workerJob->partitionKeyValue != NULL)
+		{
+			partitionColumnValue = workerJob->partitionKeyValue->constvalue;
+			partitionColumnType = workerJob->partitionKeyValue->consttype;
+			partitionColumnString = DatumToString(partitionColumnValue,
+												  partitionColumnType);
+		}
```

Comment:
maybe wrap in something like `task->partitionColumnValueString = PartitionColumnValueStringFromJob(workerJob)`  since this seems to repeat in a few places

Reply:
related, do we need to do the string conversion at this particular moment, or could we do it more lazily?

Reply:
I moved it to end functions